app/models/personal_access_token.rb

```diff
@@ -76,4 +78,21 @@ def active?
   def used?
     last_used_at.present?
   end
+
+  def expires_at=(value)
+    @expires_at_raw = value
+    super
+  end
+
+  def valid_expires_at
+    if @expires_at_raw && attributes['expires_at'].nil?
+      errors.add(:expires_at, "Could not parse timestamp '#{@expires_at_raw}'")
+    end
+  end
+
+  def expires_at_in_future
+    if changes.key?('expires_at') && expires_at.present? && expires_at < Time.zone.today
```

Comment:
Shouldn't this be "now" instead of "today?

And excuse my limited Rails knowledge, but why does this use `expires_at` while the other uses `attributes['expires_at']`?

Reply:
> Shouldn't this be "now" instead of "today?

It indeed should.

> And excuse my limited Rails knowledge, but why does this use expires_at while the other uses attributes['expires_at']?

I'm a bit rusty with rails as well. One was a remnant of a previous approach